### DISCUSSION TITLE
[GReader API] mark-all-as-read: mystery of the 'ts' param

### DISCUSSION THREAD

**Discussion by bhj** [FreshRSS/FreshRSS#7117] [Q&A] [answered]
Hi again @Alkarex :)

Very debatable whether this is a bug, but where the `ts` param is actually described, it seems to indicate that only timestamps *older* than the one provided should be marked:

- https://feedhq.readthedocs.io/en/latest/api/reference.html#mark-all-as-read
- https://github.com/theoldreader/api#marking-all-as-read

However, currently the comparison is always less than *or equal to*. What do you think?

**Comment by Alkarex** [comment, marked as answer]
Hello,
I think the `<=` makes sense here as it is used in the context of synchronisation between a client and the server.
The client knows the time of the last sync or the time information of the last received item (i.e. `crawlTimeMsec` or `timestampUsec`) and wants to mark all its local content as read. This includes the last received item, but excludes any item discovered by the server after that.

It is important to note that for this mechanism to be robust, the time comparison must be done on the time the items were added to the database (i.e. `crawlTimeMsec` or `timestampUsec`) and **not** the publication date (`published`), which may be far in the past or in the future and would lead to situations when articles are marked as read when they should not (not yet seen by the client) or not marked as read as they should.

Although less important, it is also advisable for clients to use the newest `crawlTimeMsec` or `timestampUsec` and not use their local time, as there might be time differences between the client and the server.

**Comment by bhj** [comment]
Thanks for the explanation; I think that lines up with this too: https://www.inoreader.com/developers/mark-all-as-read
